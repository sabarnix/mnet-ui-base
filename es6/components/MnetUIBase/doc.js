import { describe, PropTypes } from 'react-desc';
import { backgroundDoc, getAvailableAtBadge } from '../../utils';
export var doc = function doc(MnetUIBase) {
  var DocumentedMnetUIBase = describe(MnetUIBase).availableAt(getAvailableAtBadge('MnetUIBase')).description('The top level MnetUIBase container.').usage("import { MnetUIBase } from 'mnet-ui-base';\n<MnetUIBase>...</MnetUIBase>").intrinsicElement('div');
  DocumentedMnetUIBase.propTypes = {
    background: backgroundDoc,
    dir: PropTypes.oneOf(['rtl']).description('Layout direction for right to left contexts'),
    full: PropTypes.bool.description('Whether to take the whole viewport.').defaultValue(false),
    plain: PropTypes.bool.description("Whether or not MnetUIBase should apply a global font-family, font-size,\n        and line-height.").defaultValue(false),
    cssVars: PropTypes.bool.description('Whether to expose the css variables.').defaultValue(false),
    theme: PropTypes.object.description('Custom styles for MnetUIBase app component.'),
    themeMode: PropTypes.oneOf(['dark', 'light']).description("Dark vs. light theme variation. Default is unspecified and left to\n      theme."),
    userAgent: PropTypes.string.description("User agent used to detect the device width for setting the initial\n      breakpoint.")
  };
  return DocumentedMnetUIBase;
};
export var themeDoc = {
  'mnet.extend': {
    description: 'Any additional style for MnetUIBase.',
    type: 'string | (props) => {}',
    defaultValue: undefined
  },
  'global.font.face': {
    description: 'Custom font face declaration',
    type: 'string | (props) => {}',
    defaultValue: undefined
  }
};