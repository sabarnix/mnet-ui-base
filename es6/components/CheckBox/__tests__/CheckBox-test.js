import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { MnetUIBase } from '../../MnetUIBase';
import { CheckBox } from '..';
test('label renders', function () {
  var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(CheckBox, {
    label: "test label"
  }), /*#__PURE__*/React.createElement(CheckBox, {
    label: /*#__PURE__*/React.createElement("div", null, "test label")
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('checked renders', function () {
  var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(CheckBox, {
    checked: true
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('disabled renders', function () {
  var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(CheckBox, {
    disabled: true
  }), /*#__PURE__*/React.createElement(CheckBox, {
    disabled: true,
    checked: true
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('reverse renders', function () {
  var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(CheckBox, {
    reverse: true,
    label: "test label"
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('toggle renders', function () {
  var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(CheckBox, {
    toggle: true
  }), /*#__PURE__*/React.createElement(CheckBox, {
    toggle: true,
    checked: true
  }), /*#__PURE__*/React.createElement(CheckBox, {
    toggle: true,
    label: "test label"
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('indeterminate renders', function () {
  var component = renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(CheckBox, {
    indeterminate: true
  }), /*#__PURE__*/React.createElement(CheckBox, {
    indeterminate: true,
    label: "test label"
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('indeterminate checked warns', function () {
  var spy = jest.spyOn(global.console, 'warn');
  renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(CheckBox, {
    indeterminate: true,
    checked: true
  })));
  expect(spy).toBeCalledWith('Checkbox cannot be "checked" and "indeterminate" at the same time.');
});
test('indeterminate toggle warns', function () {
  var spy = jest.spyOn(global.console, 'warn');
  renderer.create( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(CheckBox, {
    indeterminate: true,
    toggle: true
  })));
  expect(spy).toBeCalledWith('Checkbox of type toggle does not have "indeterminate" state.');
});
test('controlled', function () {
  var _render = render( /*#__PURE__*/React.createElement(MnetUIBase, null, /*#__PURE__*/React.createElement(CheckBox, {
    label: "test-label",
    checked: true
  }))),
      container = _render.container,
      getByText = _render.getByText;

  expect(container.firstChild).toMatchSnapshot();
  fireEvent.click(getByText('test-label'));
  expect(container.firstChild).toMatchSnapshot();
});