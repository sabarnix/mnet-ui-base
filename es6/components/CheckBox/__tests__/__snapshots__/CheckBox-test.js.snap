// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checked renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 2px #E15151;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 16px;
  width: 16px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
}

.c6 {
  box-sizing: border-box;
  stroke-width: 4px;
  stroke: #E15151;
  width: 16px;
  height: 16px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
}

.c1:hover input:not([disabled]) + div,
.c1:hover input:not([disabled]) + span {
  border-color: #000000;
}

.c4 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0 !important;
  cursor: pointer;
}

.c4:checked + span > span {
  left: calc( 32px - 16px );
  background: #d9d9d9;
}

.c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

@media only screen and (max-width:512px) {
  .c5 {
    border: solid 2px #E15151;
  }
}

<div
  className="c0"
>
  <label
    checked={true}
    className="c1"
    onClick={[Function]}
  >
    <div
      checked={true}
      className="c2 c3"
    >
      <input
        checked={true}
        className="c4"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <div
        checked={true}
        className="c5 "
      >
        <svg
          checked={true}
          className="c6"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <path
            d="M6,11.3 L10.3,16 L18,6.2"
            fill="none"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`controlled 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-right: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 2px #E15151;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 16px;
  width: 16px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
}

.c6 {
  box-sizing: border-box;
  stroke-width: 4px;
  stroke: #E15151;
  width: 16px;
  height: 16px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
}

.c1:hover input:not([disabled]) + div,
.c1:hover input:not([disabled]) + span {
  border-color: #000000;
}

.c4 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0 !important;
  cursor: pointer;
}

.c4:checked + span > span {
  left: calc( 32px - 16px );
  background: #d9d9d9;
}

.c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

@media only screen and (max-width:512px) {
  .c2 {
    margin-right: 4px;
  }
}

@media only screen and (max-width:512px) {
  .c5 {
    border: solid 2px #E15151;
  }
}

<div
  class="c0"
>
  <label
    class="c1"
  >
    <div
      class="c2 c3"
    >
      <input
        checked=""
        class="c4"
        type="checkbox"
      />
      <div
        class="c5 "
      >
        <svg
          class="c6"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <path
            d="M6,11.3 L10.3,16 L18,6.2"
            fill="none"
          />
        </svg>
      </div>
    </div>
    <span>
      test-label
    </span>
  </label>
</div>
`;

exports[`controlled 2`] = `
<div
  class="StyledMnetUIBase-sc-17lh5y2-0 HlCER"
>
  <label
    class="StyledCheckBox__StyledCheckBoxContainer-sc-14qttcc-1 hcmdkc"
  >
    <div
      class="StyledBox-sc-1etxkdl-0 cTUKhN StyledCheckBox-sc-14qttcc-6 bJyswm"
    >
      <input
        checked=""
        class="StyledCheckBox__StyledCheckBoxInput-sc-14qttcc-2 frRBfI"
        type="checkbox"
      />
      <div
        class="StyledBox-sc-1etxkdl-0 cxCeJz StyledCheckBox__StyledCheckBoxBox-sc-14qttcc-3 fHlFPM"
      >
        <svg
          class="StyledCheckBox__StyledCheckBoxIcon-sc-14qttcc-0 ia-dhfe"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <path
            d="M6,11.3 L10.3,16 L18,6.2"
            fill="none"
          />
        </svg>
      </div>
    </div>
    <span>
      test-label
    </span>
  </label>
</div>
`;

exports[`disabled renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 2px rgba(0,0,0,0.15);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 16px;
  width: 16px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 2px #E15151;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 16px;
  width: 16px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
}

.c7 {
  box-sizing: border-box;
  stroke-width: 4px;
  stroke: #E15151;
  width: 16px;
  height: 16px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  opacity: 0.5;
  cursor: default;
}

.c1:hover input:not([disabled]) + div,
.c1:hover input:not([disabled]) + span {
  border-color: #000000;
}

.c4 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0 !important;
}

.c4:checked + span > span {
  left: calc( 32px - 16px );
  background: #d9d9d9;
}

.c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

@media only screen and (max-width:512px) {
  .c5 {
    border: solid 2px rgba(0,0,0,0.15);
  }
}

@media only screen and (max-width:512px) {
  .c6 {
    border: solid 2px #E15151;
  }
}

<div
  className="c0"
>
  <label
    checked={false}
    className="c1"
    disabled={true}
    onClick={[Function]}
  >
    <div
      checked={false}
      className="c2 c3"
      disabled={true}
    >
      <input
        checked={false}
        className="c4"
        disabled={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <div
        checked={false}
        className="c5 "
        disabled={true}
      />
    </div>
  </label>
  <label
    checked={true}
    className="c1"
    disabled={true}
    onClick={[Function]}
  >
    <div
      checked={true}
      className="c2 c3"
      disabled={true}
    >
      <input
        checked={true}
        className="c4"
        disabled={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <div
        checked={true}
        className="c6 "
        disabled={true}
      >
        <svg
          checked={true}
          className="c7"
          disabled={true}
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <path
            d="M6,11.3 L10.3,16 L18,6.2"
            fill="none"
          />
        </svg>
      </div>
      <input
        type="hidden"
        value="true"
      />
    </div>
  </label>
</div>
`;

exports[`indeterminate renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 2px rgba(0,0,0,0.15);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 16px;
  width: 16px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-right: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  box-sizing: border-box;
  stroke-width: 4px;
  stroke: #E15151;
  width: 16px;
  height: 16px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
}

.c1:hover input:not([disabled]) + div,
.c1:hover input:not([disabled]) + span {
  border-color: #000000;
}

.c4 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0 !important;
  cursor: pointer;
}

.c4:checked + span > span {
  left: calc( 32px - 16px );
  background: #d9d9d9;
}

.c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

@media only screen and (max-width:512px) {
  .c5 {
    border: solid 2px rgba(0,0,0,0.15);
  }
}

@media only screen and (max-width:512px) {
  .c7 {
    margin-right: 4px;
  }
}

<div
  className="c0"
>
  <label
    checked={false}
    className="c1"
    onClick={[Function]}
  >
    <div
      checked={false}
      className="c2 c3"
    >
      <input
        checked={false}
        className="c4"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <div
        checked={false}
        className="c5 "
      >
        <svg
          checked={false}
          className="c6"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <path
            d="M6,12 L18,12"
            fill="none"
          />
        </svg>
      </div>
    </div>
  </label>
  <label
    checked={false}
    className="c1"
    onClick={[Function]}
  >
    <div
      checked={false}
      className="c7 c3"
    >
      <input
        checked={false}
        className="c4"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <div
        checked={false}
        className="c5 "
      >
        <svg
          checked={false}
          className="c6"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <path
            d="M6,12 L18,12"
            fill="none"
          />
        </svg>
      </div>
    </div>
    <span>
      test label
    </span>
  </label>
</div>
`;

exports[`label renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-right: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 2px rgba(0,0,0,0.15);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 16px;
  width: 16px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
}

.c1:hover input:not([disabled]) + div,
.c1:hover input:not([disabled]) + span {
  border-color: #000000;
}

.c4 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0 !important;
  cursor: pointer;
}

.c4:checked + span > span {
  left: calc( 32px - 16px );
  background: #d9d9d9;
}

.c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

@media only screen and (max-width:512px) {
  .c2 {
    margin-right: 4px;
  }
}

@media only screen and (max-width:512px) {
  .c5 {
    border: solid 2px rgba(0,0,0,0.15);
  }
}

<div
  className="c0"
>
  <label
    checked={false}
    className="c1"
    onClick={[Function]}
  >
    <div
      checked={false}
      className="c2 c3"
    >
      <input
        checked={false}
        className="c4"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <div
        checked={false}
        className="c5 "
      />
    </div>
    <span>
      test label
    </span>
  </label>
  <label
    checked={false}
    className="c1"
    onClick={[Function]}
  >
    <div
      checked={false}
      className="c2 c3"
    >
      <input
        checked={false}
        className="c4"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <div
        checked={false}
        className="c5 "
      />
    </div>
    <div>
      test label
    </div>
  </label>
</div>
`;

exports[`reverse renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-left: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 2px rgba(0,0,0,0.15);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 16px;
  width: 16px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
}

.c1:hover input:not([disabled]) + div,
.c1:hover input:not([disabled]) + span {
  border-color: #000000;
}

.c4 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0 !important;
  cursor: pointer;
}

.c4:checked + span > span {
  left: calc( 32px - 16px );
  background: #d9d9d9;
}

.c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

@media only screen and (max-width:512px) {
  .c2 {
    margin-left: 4px;
  }
}

@media only screen and (max-width:512px) {
  .c5 {
    border: solid 2px rgba(0,0,0,0.15);
  }
}

<div
  className="c0"
>
  <label
    checked={false}
    className="c1"
    onClick={[Function]}
  >
    <span>
      test label
    </span>
    <div
      checked={false}
      className="c2 c3"
    >
      <input
        checked={false}
        className="c4"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <div
        checked={false}
        className="c5 "
      />
    </div>
  </label>
</div>
`;

exports[`toggle renders 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-right: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
}

.c1:hover input:not([disabled]) + div,
.c1:hover input:not([disabled]) + span {
  border-color: #000000;
}

.c4 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0 !important;
  cursor: pointer;
}

.c4:checked + span > span {
  left: calc( 32px - 16px );
  background: #d9d9d9;
}

.c5 {
  box-sizing: border-box;
  vertical-align: middle;
  display: inline-block;
  width: 32px;
  height: 16px;
  border: 2px solid;
  border-color: rgba(0,0,0,0.15);
  border-radius: 16px;
  background-color: transparent;
}

.c6 {
  box-sizing: border-box;
  position: relative;
  display: inherit;
  top: -2px;
  left: -2px;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  width: 16px;
  height: 16px;
  background: #d9d9d9;
  border-radius: 16px;
}

.c3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

@media only screen and (max-width:512px) {
  .c7 {
    margin-right: 4px;
  }
}

<div
  className="c0"
>
  <label
    checked={false}
    className="c1"
    onClick={[Function]}
  >
    <div
      checked={false}
      className="c2 c3"
    >
      <input
        checked={false}
        className="c4"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <span
        checked={false}
        className="c5"
      >
        <span
          checked={false}
          className="c6"
        />
      </span>
    </div>
  </label>
  <label
    checked={true}
    className="c1"
    onClick={[Function]}
  >
    <div
      checked={true}
      className="c2 c3"
    >
      <input
        checked={true}
        className="c4"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <span
        checked={true}
        className="c5"
      >
        <span
          checked={true}
          className="c6"
        />
      </span>
    </div>
  </label>
  <label
    checked={false}
    className="c1"
    onClick={[Function]}
  >
    <div
      checked={false}
      className="c7 c3"
    >
      <input
        checked={false}
        className="c4"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <span
        checked={false}
        className="c5"
      >
        <span
          checked={false}
          className="c6"
        />
      </span>
    </div>
    <span>
      test label
    </span>
  </label>
</div>
`;
